File: old_backend/important-core-code/python/vulnerable_sql.py
# INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
# This file contains critical SQL injection vulnerabilities for testing Aikido scanner

import sqlite3
from flask import Flask, request

app = Flask(__name__)

@app.route('/user')
def get_user():
    user_id = request.args.get('id')
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    # VULNERABLE: String formatting in SQL
    query = f"SELECT * FROM users WHERE id = '{user_id}'"
    cursor.execute(query)
    return cursor.fetchall()

@app.route('/login', methods=['POST'])
def login():
    username = request.form.get('username')
    password = request.form.get('password')
    # VULNERABLE: String concatenation
    query = "SELECT * FROM users WHERE username = '" + username + "' AND password = '" + password + "'"
    cursor.execute(query)
    return cursor.fetchone()

if __name__ == '__main__':
    app.run()

File: old_backend/important-core-code/java/VulnerableSQL.java
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical SQL injection vulnerabilities for testing Aikido scanner

import java.sql.*;
import javax.servlet.http.*;

public class VulnerableSQL extends HttpServlet {
    
    public void doGet(HttpServletRequest request, HttpServletResponse response) {
        String userId = request.getParameter("id");
        
        try {
            Connection conn = DriverManager.getConnection(
                "jdbc:mysql://localhost/db", "root", "password123"
            );
            Statement stmt = conn.createStatement();
            
            // VULNERABLE: String concatenation
            String query = "SELECT * FROM users WHERE id = '" + userId + "'";
            ResultSet rs = stmt.executeQuery(query);
            
            // VULNERABLE: String format
            String search = request.getParameter("search");
            String query2 = String.format("SELECT * FROM products WHERE name LIKE '%%%s%%'", search);
            rs = stmt.executeQuery(query2);
            
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}
